webapp-django/crashstats/api/views.py

```diff
@@ -463,8 +458,27 @@ def dedent_left(text, spaces):
     return "\n".join(lines)
 
 
+def handle_ratelimit(fun):
+    """Fixes django-ratelimit exception so it ends up as a 429
+
+    Note: This must be before ratelimit so it can handle exceptions.
+
+    """
+
+    @wraps(fun)
+    def _handle_ratelimit(*args, **kwargs):
+        try:
+            return fun(*args, **kwargs)
+        except Ratelimited:
+            # If the view is rate limited, we throw a 429.
+            raise Throttled()
+
+    return _handle_ratelimit
```

Comment:
ratelimit returns a 403 which is wrong for our uses, so this wraps views and swaps a 403 for a 429.